doc/source/api/metadata.rst

```diff
@@ -10,20 +10,117 @@ Goals and Scope
+name                      a more descriptive object label or legacy ID; should *not* be used as reference
+accessions                an ``ARRAY`` containing other names, IDs, URIs of the object
+description               a string describing aspects of the object; *not* a list or nested object itself
+recordCreateTime          time at which this record was created in ISO 8601 (see **Date and Time...**)
+recordUpdateTime          time at which this record was updated in ISO 8601 (see **Date and Time...**)
+========================= ======================================================
+
+Date and Time Format Specifications
+@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+
+Date and time formats are specified as ISO8601 compatible strings, both for
+time points as well as for intervals and durations.
+The required granularity has to be specified as part of the respective
+attributes' documentations.
+
+===========
+Time points
+===========
+
+The specification of a time point is given through the concatenation of
+
+* a date in YYYY-MM-DD
+* the designator "T" indicating a following time description
+* the time of day in HH:MM:SS.SSS form, where "SSS" represents a decimal fraction of a second
+* a time zone offset in relation to UTC
+
+**Examples**
+
+* year (YYYY)
+    2015
+
+* date (e.g. date of birth) in YYYY-MM-DD
+    2015-02-10
+
+* time stamp in milliseconds in YYYY-MM-DDTHH:MM:SS.SSS
+    2015-02-10T00:03:42.123Z
+
+**Implementations**
+
+* recordCreateTime (G2P, metadata)
+* recordUpdateTime (G2P, metadata)
+* many proposed in the metadata branch
+
+===========
+Durations
+===========
+
+Durations are a specific form of intervals, without reference to time points.
+They are indicated with a leading "P", followed by unit delimited
+quantifiers. A leading "T" is required before the start of the time components.
+Durations do not have to be normalized; "PT50H" is equally valid as "P2T2H".
+
+**Examples**
+
+* age in years in PnY
+    P44Y
+
+* age in years and months in PnYnM
+    P44Y08M
+
+* short time interval (e.g. 30min in experimental time series) in PTnM
+    PT30M
+
+==============
+Time intervals
+==============
+
+Time intervals consist of a combination of two time designators. These can be
+either two time points for start and end, or one time point and a leading
+(time point indicates end) or trailing (time point indicates start) duration.
+The time elements are separated by a forward slash "/".
+
+**Examples**
+
+* age with date of birth in YYYY-MM-DD/PnYnMnD
+    1967-11-21/P40Y10M05D
+
+* anchored 3 month interval, e.g. a therapy cycle in YYYY-MM-DD/YYYY-MM-DD
+    2015-04-18/2015-07-17
+
+* experimental intervention of 30min in YYYY-MM-DDTHH:MM/YYYY-MM-DDTHH:MM
+    2014-12-31T23H45M/2015-01-01T00H15M
+
+
+Object: Dataset
+@@@@@@@@@@@@@@@
+
+GA4GH data objects can be part of one or several *dataset*(s). A dataset is a
 data-provider-specified collection of related data of multiple types.
```

Comment:
We should setup a topic branch for bring in new metadata
objects, as we need to implement them in the server and
compliance suite before moving to master.  maybe metadata_stage ?

Michael Baudis <notifications@github.com> writes:
> In doc/source/api/metadata.rst:
> 
> > +**Examples**
> > +
> > +* age with date of birth in YYYY-MM-DD/PnYnMnD
> > +    1967-11-21/P40Y10M05D
> > +
> > +* anchored 3 month interval, e.g. a therapy cycle in YYYY-MM-DD/YYYY-MM-DD
> > +    2015-04-18/2015-07-17
> > +
> > +* experimental intervention of 30min in YYYY-MM-DDTHH:MM/YYYY-MM-DDTHH:MM
> > +    2014-12-31T23H45M/2015-01-01T00H15M
> > +
> > +
> > +Object: Dataset
> > +@@@@@@@@@@@@@@@
> > +
> > +GA4GH data objects can be part of one or several *dataset*(s). A dataset is a
> >  data-provider-specified collection of related data of multiple types.
> 
> I’ve tried to fix the BioSample PR, by fixing reads/variants (which were
> obviously erroneous - why didn’t anybody take care of this???). #397 This was
> supported before, only didn’t get in due to your objections since it wasn’t
> “bioSampleId” etc. => ??? Michael.
> 
> —
> Reply to this email directly or view it on GitHub.*
>